pandas/core/arrays/arrow/dtype.py

```diff
@@ -8,28 +8,42 @@
 
 from pandas.core.dtypes.base import StorageExtensionDtype
 
-from pandas.core.arrays.arrow import ArrowExtensionArray
-
 
 class ArrowDtype(StorageExtensionDtype):
     """
-    Base class for dtypes for BaseArrowArray subclasses.
+    Base class for dtypes for ArrowExtensionArray.
     Modeled after BaseMaskedDtype
     """
 
-    name: str
-    base = None
-    type: pa.DataType
-
     na_value = pa.NA
 
-    def __init__(self, storage="pyarrow") -> None:
-        super().__init__(storage)
+    def __init__(self, pa_dtype: pa.DataType) -> None:
+        super().__init__("pyarrow")
+        if not isinstance(pa_dtype, pa.DataType):
+            raise ValueError(
+                f"pa_dtype ({pa_dtype}) must be an instance "
+                f"of a pyarrow.DataType. Got {type(pa_dtype)} instead."
+            )
+        self.pa_dtype = pa_dtype
+
+    @property
+    def type(self):
+        """
+        The scalar type for the array, e.g. ``int``
+        """
+        return type(self.pa_dtype)
```

Comment:
Ah good point. So in pyarrow, the actual data type is an attribute of a `pyarrow.DataType` correct? i.e. it might not be possible to return a `type` that refects the specific data type for pyarrow?

Reply:
Actually for internal use cases, this is okay that it just returns `pyarrow.DataType` I think. It's just used in `__eq__` where also the name of the `self.pyarrow_type` instance is checked as well

Reply:
I took a look at where this was introduced in `__eq__` (and commented about it on https://github.com/pandas-dev/pandas/pull/46537), but so I _think_ that we should actually be harcoding `str` in `__eq__` instead of `self.type`. Which would remove our internal usage of the attribute.

I don't know if we actually use `dtype.type` anywhere internally. AFAIK we mostly added this to the ExtensionDtype originally for consistency / compatibility with numpy dtypes. 
For numpy, this returns the scalar type:

```
>>> np.dtype("int64")
dtype('int64')
>>> np.dtype("int64").type
numpy.int64
>>> type(np.array([1, 2])[0])
numpy.int64
```

So that in theory `isinstance(arr[0], arr.dtype.type)` holds. 

But again I don't know if we actually rely on that behaviour ourselves (in theory we could maybe use it for something like https://github.com/pandas-dev/pandas/issues/27462), or if any user of pandas relies on that behaviour.